includes/Core/Storage/Data_Encryption.php

```diff
@@ -83,19 +83,23 @@ public function encrypt( $value ) {
 	 * @return string|bool Decrypted value, or false on failure.
 	 */
 	public function decrypt( $raw_value ) {
-		if ( ! extension_loaded( 'openssl' ) ) {
+		if ( ! extension_loaded( 'openssl' ) || ! is_string( $raw_value ) ) {
 			return $raw_value;
 		}
 
-		$raw_value = base64_decode( $raw_value, true );
+		$decoded_value = base64_decode( $raw_value, true );
+
+		if ( false === $decoded_value ) {
+			return $raw_value;
+		}
```

Comment:
Would it make sense to at least log an error or debug message here? Running into this would indicate we have a value that we _expect_ to be encrypted, but it isn't, so we probably shouldn't silently fail. At least in development, this would be a useful warning to flag.

```suggestion
		$decoded_value = base64_decode( $raw_value, true );

		if ( false === $decoded_value ) {
			return $raw_value;
		}
```

Reply:
@tofumatt Thanks, this is really a good point